### PR TITLE
Add Java Setup to all release jobs

### PR DESCRIPTION
**Pull request by tdchow**
### Summary of changes

 - Add Setup Java steps to release workflow
 - Add Temp Release Workflow for running `bump_version` and `create_github_release` jobs for `5.0.0-beta1` release. This workflow will be removed once it runs.

### Checklist

 - [ ] Added a changelog entry
 - [ ] Relevant test coverage

### CODE REVIEW

File: .github/workflows/temp_release.yml [hunk 1]
```diff
@@ -0,0 +1,59 @@
+name: Temp Release Workflow
```
Comment (jaxdesmarais) [inline .github/workflows/temp_release.yml:1]: We can also run these steps manually vs creating a removing a new workflow. That's what we've typically done previously on iOS at least unless some of these steps cannot be done manually.

Comment (tdchow) [reply .github/workflows/temp_release.yml:1]: We don't currently have that option with how the workflow is set up. Right now each job requires the previous one: https://github.com/braintree/braintree_android/blob/e5500e0f270b7fd14e66eee19ec461c262ad1cef/.github/workflows/release.yml#L249

I feel that there's a benefit in restricting the running of individual jobs and enforcing the entire workflow to be run for a release. I worry that it'd open up the possibility of publishing something to maven central by accident.

Comment (jaxdesmarais) [reply .github/workflows/temp_release.yml:1]: iOS does not rely on the same settings, really the [iOS releases](https://github.com/braintree/braintree_ios/blob/main/.github/workflows/release.yml) as a whole are pretty different than what we do on android (for example we don't run all the tests again). It is nice in that say cocoapods or publishing reference docs fails we typically just run those 1-2 steps manually. We do rely on the order of things but can pick up where the release failed if something goes sideways.

**Review by sarahkoop — APPROVED** [review]

File: .github/workflows/temp_release.yml (hunk 1 above)
Comment (tdchow) [reply .github/workflows/temp_release.yml:1]: Gotcha. I think the Android release workflow has been stable up until v5, so hopefully we won't have any of these CI issues until the next major version where our gradle version and build tools are bumped.

Are you okay with merging in the temp workflow, then following up with another PR to remove it?

**Review by scannillo — APPROVED** [review]
LGTM! Maybe leaving a comment in the new temp workflow to eventually remove will help us remember

**Review by jaxdesmarais — APPROVED** [review]

**Comment by tdchow** [comment]
> LGTM! Maybe leaving a comment in the new temp workflow to eventually remove will help us remember

I'll plan to open another PR for removing the workflow once it runs successfully. So it shouldn't exist for very long!

**Pull request merged**